algorithms/src/crh/mod.rs

```diff
@@ -26,9 +26,6 @@ pub use pedersen::*;
 pub mod pedersen_compressed;
 pub use pedersen_compressed::*;
 
-pub mod poseidon;
-pub use poseidon::*;
```

Comment:
We drop the Poseidon CRH. It seems that the current implementation did not enforce `INPUT_SIZE_BITS` and simply truncates the part after it in the gadgets, so there is a security issue. Given that we are not using Poseidon CRH for now, I prefer to drop this implementation.

Reply:
Actually we need this upstream, currently using it for a Merkle tree construction on the application level